Userland/Libraries/LibWeb/Infra/ByteSequences.cpp

```diff
@@ -25,4 +25,35 @@ void byte_uppercase(ByteBuffer& bytes)
         bytes[i] = to_ascii_uppercase(bytes[i]);
 }
 
+// https://infra.spec.whatwg.org/#byte-sequence-prefix
+// https://infra.spec.whatwg.org/#byte-sequence-starts-with
+bool is_prefix_of(ByteBuffer const& potential_prefix, ByteBuffer const& input)
+{
+    // "input starts with potentialPrefix" can be used as a synonym for "potentialPrefix is a prefix of input".
+    return input.span().starts_with(potential_prefix.span());
+}
+
+// https://infra.spec.whatwg.org/#byte-less-than
+bool is_byte_less_than(ByteBuffer const& a, ByteBuffer const& b)
+{
+    // 1. If b is a prefix of a, then return false.
+    if (is_prefix_of(b, a))
+        return false;
+
+    // 2. If a is a prefix of b, then return true.
+    if (is_prefix_of(a, b))
+        return true;
+
+    // 3. Let n be the smallest index such that the nth byte of a is different from the nth byte of b.
+    //    (There has to be such an index, since neither byte sequence is a prefix of the other.)
+    // 4. If the nth byte of a is less than the nth byte of b, then return true.
+    // 5. Return false.
+    for (size_t i = 0; i < a.size(); ++i) {
+        VERIFY(i < b.size());
```

Comment:
wouldnʼt it be better if you `VERIFY(a.size() < b.size())` (or even `<=`) before the loop? 🤔 (still not convinced; that is already handled in `is_prefix_of`/`starts_with`.)